src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -49,9 +48,15 @@ trait RegularSync {
       waitingForActor = None
       handleBlockBodies(peer, blockBodies)
 
-    case block: BroadcastBlocks if broadcasting =>
+    case BroadcastBlocks(newBlocks) if broadcasting =>
       //FIXME: Decide block propagation algorithm (for now we send block to every peer) [EC-87]
-      peersToDownloadFrom.keys.foreach(_.ref ! block)
+      val blocksToSendToEachPeer = for {
+        peer <- peersToDownloadFrom.keys
```

Comment:
That makes sense :+1:
I'm wondering, what do you think about solving this by adding a `PeerConnection` event to the `PeerEventBus` so that the `peersToDownloadFrom` is up-to-date? As you suggested this would be a nice optimization instead of periodically asking for peers in the `SyncController` and it seems to solve this issue.